napari_pyclesperanto_assistant/_gui/_category_widget.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import annotations
+
+from inspect import Parameter, Signature, signature
+from typing import TYPE_CHECKING, Sequence, Tuple
+
+import pyclesperanto_prototype as cle
+from loguru import logger
+from magicgui import magicgui
+from typing_extensions import Annotated
+
+from .._categories import Category
+
+if TYPE_CHECKING:
+    from napari.layers import Layer
+
+VIEWER_PARAM = "viewer"
+OP_NAME_PARAM = "op_name"
+OP_ID = "op_id"
+
+
+def num_positional_args(func, types=[cle.Image, int, str, float, bool]) -> int:
+    params = signature(func).parameters
+    return len([p for p in params.values() if p.annotation in types])
+
+
+# TODO
+# def _label_parameters(operation, parameters):
+#     sig = signature(operation)
+
+#     count = 0
+#     for parameter in sig.parameters:
+#         if (
+#             sig.parameters[parameter].annotation in [int, str, float, bool]
+#             and len(parameters) > count
+#         ):
+#             parameters[count].label = parameter
+#             parameters[count].text = parameter
+#             count += 1
+#     for i in range(count, len(parameters)):
```

Comment:
Here I couldn't figure out if it's possible to hide GUI-elements from the magicgui-dialog in case the operation doesn't have this parameter. E.g. in the binarize category, there exist operations (e.g. `greater_constant`) which have one numeric parameter and others (e.g. `threshold_otsu`) which have none.

Reply:
got this figured out